src/Panel.tsx

```diff
@@ -3,28 +3,28 @@ import { useChannel } from "@storybook/api";
 import { AddonPanel } from "@storybook/components";
 import { EVENTS } from "./constants";
 import { PanelContent } from "./components/PanelContent";
-import { STORY_CHANGED } from "@storybook/core-events";
+import StorybookCoreEvents from "@storybook/core-events";
+import { ChangeOptions } from "./withRoundTrip";
+
+const { STORY_CHANGED } = StorybookCoreEvents;
 
 interface PanelProps {
   active: boolean;
 }
 
-const INITIAL_STATE = {
-  // @ts-ignore
-  // TODO: add typing
-  changes: [],
+type State = {
+  changes: ChangeOptions[];
 };
 
-interface State {
-  changes: any[]; // TODO: add shared Change type
-}
+const INITIAL_STATE: State = {
+  changes: [],
+};
 
-interface Action {
-  type: string; // TODO: enumerate possible actions
-  payload?: unknown; // TODO: use shared Change type
-}
+type KnownActions =
```

Comment:
I love this notion of specifying `KnownActions`. That is a really great tip. Is this a standard or a practice that you use in the OSS projects you work in?

Reply:
This is a common pattern for Redux in the wild... but we actually recommend against it. In this case for this addon, it's a non-issue. Background: https://phryneas.de/redux-typescript-no-discriminating-union

Reply:
awesome. thanks so much for the follow-up that. this is a good read.